models/migrations/v163.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"xorm.io/xorm"
+)
+
+func convertTopicNameFrom25To50(x *xorm.Engine) error {
+	type Topic struct {
+		ID   int64  `xorm:"pk autoincr"`
+		Name string `xorm:"UNIQUE VARCHAR(50)"`
+	}
+	if err := x.Sync2(new(Topic)); err != nil {
+		return err
+	}
+
+	sess := x.NewSession()
+	defer sess.Close()
+	if err := sess.Begin(); err != nil {
+		return err
+	}
+	if err := recreateTable(sess, new(Topic)); err != nil {
```

Comment:
**// WARNING: YOU MUST PROVIDE THE FULL BEAN DEFINITION**

Reply:
the `type Topic struct` has to be a full representation of Topic as you need from this point onwards.